concordium-consensus/src/Concordium/Skov/Update.hs

```diff
@@ -461,39 +461,70 @@ doStoreBlock pb@GB.PendingBlock{..} = unlessShutDown $ do
 --   * 'ResultShutDown' which indicates that consensus was shut down, and so the transaction was not added.
 doReceiveTransaction :: (TreeStateMonad pv m, TimeMonad m, SkovQueryMonad pv m) => BlockItem -> Slot -> m UpdateResult
 doReceiveTransaction tr slot = unlessShutDown $ do
-  (_, ur) <- doReceiveTransactionInternal tr slot
-  when (ur == ResultSuccess) $ purgeTransactionTable False =<< currentTime
-  return ur
+  -- Don't accept the transaction if there are too many pending transactions.
+  pendingNum <- numPendingTransactions <$> getPendingTransactions
+  maxPendingNum <- rpMaxPendingTransactionNum <$> getRuntimeParameters
+  if pendingNum > maxPendingNum
+  then return ResultUnverifiable
```

Comment:
I think this is abusing this return value. We should add a new return code for it.